File: TransCoder_tokenized_test_set_functions/python/PRINT_POSSIBLE_EDGES_TREE_GIVEN_DIAMETER_HEIGHT_VERTICES.py
def constructTree ( n, d, h ) :
    if d == 1 :
        if n == 2 and h == 1 :
            print ( "1 2" )
            return 0
        print ( "-1" )
        return 0
    if d > 2 * h :
        print ( "-1" )
        return 0
    for i in range ( 1, h + 1 ) :
        print ( i, " ", i + 1 )
    if d > h :
        print ( 1, "  ", h + 2 )
        for i in range ( h + 2, d + 1 ) :
            print ( i, " ", i + 1 )
    for i in range ( d + 1, n ) :
        k = 1
        if d == h :
            k = 2
        print ( k, " ", i + 1 )
    

File: TransCoder_tokenized_test_set_functions/java/PRINT_POSSIBLE_EDGES_TREE_GIVEN_DIAMETER_HEIGHT_VERTICES.java
static void constructTree ( int n, int d, int h ) {
  if ( d == 1 ) {
    if ( n == 2 && h == 1 ) {
      System . out . println ( "1 2" );
      return;
    }
    System . out . println ( "-1" );
    return;
  }
  if ( d > 2 * h ) {
    System . out . println ( "-1" );
    return;
  }
  for ( int i = 1;
  i <= h;
  i ++ ) System . out . println ( i + " " + ( i + 1 ) );
  if ( d > h ) {
    System . out . println ( "1" + " " + ( h + 2 ) );
    for ( int i = h + 2;
    i <= d;
    i ++ ) {
      System . out . println ( i + " " + ( i + 1 ) );
    }
  }
  for ( int i = d + 1;
  i < n;
  i ++ ) {
    int k = 1;
    if ( d == h ) k = 2;
    System . out . println ( k + " " + ( i + 1 ) );
  }
}